doc: twister.rst rule violations blocking CI

### Describe the bug

Doc heading hierarchy violations in twister.rst are causing doc builds to fail, blocking CI.  @kartben , this issue comes from 1e8a5f7dd0c9c0d8819c2503a71b2001c2780169.

PR https://github.com/zephyrproject-rtos/zephyr/pull/99029 is blocked with this [doc build error report](https://github.com/zephyrproject-rtos/zephyr/actions/runs/19148807427/job/54733464075?pr=99029):
```
Run if [[ "$GITHUB_REF" =~ "refs/tags/v" ]]; then
Run '/home/runner/k8s/index.js'
make: Entering directory '/__w/zephyr/zephyr/doc'
make html DT_TURBO_MODE=1 HW_FEATURES_TURBO_MODE=1
make[1]: Entering directory '/__w/zephyr/zephyr/doc'
cmake \
	-GNinja \
	-B_build \
	-S. \
	-DDOC_TAG=development \
	-DSPHINXOPTS="-j 8 -W --keep-going -T" \
	-DSPHINXOPTS_EXTRA="-q -t publish" \
	-DLATEXMKOPTS="-halt-on-error -no-shell-escape" \
	-DDT_TURBO_MODE=1 \
	-DHW_FEATURES_TURBO_MODE=1 \
	-DHW_FEATURES_VENDOR_FILTER=
Loading Zephyr module(s) (Zephyr repository): doc
-- Found Python3: /opt/python/venv/bin/python (found suitable version "3.12.3", minimum required is "3.10") found components: Interpreter 
-- Cache files will be written to: /root/.cache/zephyr
-- Found west (found suitable version "1.5.0", minimum required is "1.0.0")
-- Zephyr base: /__w/zephyr/zephyr
-- Found Doxygen: /usr/local/bin/doxygen (found version "1.14.0") found components: doxygen dot 
-- Could NOT find LATEX (missing: LATEX_COMPILER PDFLATEX) 
CMake Warning at CMakeLists.txt:42 (message):
  LaTeX components not found.  PDF build will not be available.


-- Configuring done (0.6s)
-- Generating done (0.0s)
-- Build files have been written to: /__w/zephyr/zephyr/doc/_build
cmake --build _build --target html
[0/2] Generating Devicetree bindings documentation...
[1/2] Running Sphinx HTML build...
/__w/zephyr/zephyr/doc/_build/src/develop/test/twister.rst:1204: CRITICAL: Title level inconsistent:

Hardware setup
++++++++++++++ [docutils]
/__w/zephyr/zephyr/doc/_build/src/develop/test/twister.rst:1214: CRITICAL: Title level inconsistent:

Configuration
+++++++++++++ [docutils]
/__w/zephyr/zephyr/doc/_build/src/develop/test/twister.rst:1329: CRITICAL: Title level inconsistent:

Workflow
++++++++ [docutils]
FAILED: CMakeFiles/html /__w/zephyr/zephyr/doc/_build/CMakeFiles/html 
cd /__w/zephyr/zephyr/doc/_build && /usr/bin/cmake -E env DOXYGEN_EXECUTABLE=/usr/local/bin/doxygen DOT_EXECUTABLE=/usr/bin/dot DOCS_HTML_DIR=/__w/zephyr/zephyr/doc/_build/html OUTPUT_DIR=/__w/zephyr/zephyr/doc/_build/html /opt/python/venv/bin/sphinx-build -b html -c /__w/zephyr/zephyr/doc -d /__w/zephyr/zephyr/doc/_build/doctrees -w /__w/zephyr/zephyr/doc/_build/html.log -t development -t hw_features_turbo -j 8 -W --keep-going -T -q -t publish /__w/zephyr/zephyr/doc/_build/src /__w/zephyr/zephyr/doc/_build/html
ninja: build stopped: subcommand failed.
make[1]: Leaving directory '/__w/zephyr/zephyr/doc'
make[1]: *** [Makefile:26: html] Error 1
make: *** [Makefile:20: html-fast] Error 2
make: Leaving directory '/__w/zephyr/zephyr/doc'
Error: Error: failed to run script step: command terminated with non-zero exit code: error executing command [sh -e /__w/_temp/d633af40-bb4e-11f0-9dac-f179f5b3d5e8.sh], exit code 2
Error: Process completed with exit code 1.
Error: Executing the custom container implementation failed. Please contact your self hosted runner administrator.
```

### Regression

- [ ] This is a regression.

### Steps to reproduce

_No response_

### Relevant log output

```shell

```

### Impact

Showstopper – Prevents release or major functionality; system unusable.

### Environment

_No response_

### Additional Context

_No response_

--- a/doc/develop/test/twister.rst
+++ b/doc/develop/test/twister.rst
@@ -1202,7 +1202,7 @@ testing using video fingerprints.
    Window being displayed for a "compare" run where fingerprint is a 90% match with the reference.
 
 Hardware setup
-++++++++++++++
+--------------
 
 The display capture harness requires:
 
@@ -1212,7 +1212,7 @@ The display capture harness requires:
 - DUT connected to the same PC for flashing and serial console access
 
 Configuration
-+++++++++++++
+-------------
 
 The harness uses a YAML configuration file that defines camera settings, test parameters, and video
 signature analysis options. A typical configuration is shown below:
@@ -1327,7 +1327,7 @@ environment variable:
       display_capture_config: "${DISPLAY_TEST_DIR}/display_config.yaml"
 
 Workflow
-++++++++
+--------
 
 First, generate **reference fingerprints** for a known-good display output:
 

